### PR TITLE
Add mono_custom_attr_get_attrs helper

### PR DESCRIPTION
**Pull request by Tak**
I'll do a separate PR for 2017-02-staging if this change is acceptable.

### CODE REVIEW

**Comment by Tak** [comment]
FYI @amechtley

**Comment by joshpeterson** [comment]
LGTM, maybe wait for @joncham to have a look as well before merging. Also, do we need to add this to the .def file on Windows?

**Comment by Tak** [comment]
Guh, you're right, always forget about that thing.

File: mono/metadata/reflection.c [hunk 1]
```diff
@@ -8565,6 +8565,26 @@ mono_custom_attrs_has_attr (MonoCustomAttrInfo *ainfo, MonoClass *attr_klass)
...
+MonoClass*
+mono_custom_attrs_get_attrs(MonoCustomAttrInfo *ainfo, gpointer *iter)
+{
+	int index = -1;
+	if (!iter)
+		return NULL;
+	if (!*iter)
+	{
+		*iter = 1;
+		return ainfo->attrs[0].ctor->klass;
```
Comment (joncham) [inline mono/metadata/reflection.c]: Are we sure ainfo->attrs[0] is always valid?

File: mono/metadata/reflection.c [hunk 2]
```diff
@@ -8565,6 +8565,26 @@ mono_custom_attrs_has_attr (MonoCustomAttrInfo *ainfo, MonoClass *attr_klass)
...
+	}
+
+	index = (int)*iter;
```
Comment (joncham) [inline mono/metadata/reflection.c]: Please use `GPOINTER_TO_INT`

File: mono/metadata/reflection.c [hunk 3]
```diff
@@ -8565,6 +8565,26 @@ mono_custom_attrs_has_attr (MonoCustomAttrInfo *ainfo, MonoClass *attr_klass)
...
+/* Style inspired by mono_class_get_interfaces */
+MonoClass*
+mono_custom_attrs_get_attrs(MonoCustomAttrInfo *ainfo, gpointer *iter)
+{
+	int index = -1;
+	if (!iter)
+		return NULL;
+	if (!*iter)
+	{
+		*iter = 1;
```
Comment (joncham) [inline mono/metadata/reflection.c]: `GINT_TO_POINTER`

File: mono/metadata/reflection.c [hunk 4]
```diff
@@ -8565,6 +8565,26 @@ mono_custom_attrs_has_attr (MonoCustomAttrInfo *ainfo, MonoClass *attr_klass)
...
+		return ainfo->attrs[0].ctor->klass;
+	}
+
+	index = (int)*iter;
+	if (index >= ainfo->num_attrs)
+		return NULL;
+	*iter = index + 1;
```
Comment (joncham) [inline mono/metadata/reflection.c]: `GINT_TO_POINTER`

File: mono/metadata/reflection.c (hunk 1 above)
Comment (Tak) [reply mono/metadata/reflection.c]: It's the same construct used (also without checks) in e.g. `mono_custom_attrs_has_attr`

**Pull request merged**